### DISCUSSION TITLE
: ‘Buffer’ is not a member of ‘muduo::net’？

### DISCUSSION THREAD

**Discussion by sychen12345** [chenshuo/muduo#562] [General]
陈硕大佬，请问一下，
using namespace muduo;
using namespace muduo::net;
muduo::net::Buffer *buf;
        // Buffer            *buf;
        muduo::StringPiece s("buf\n");

        buf->append(s);
我这样使用muduo的buffer好像你书上也是这样写的，为什么会报错呢
 error: ‘Buffer’ is not a member of ‘muduo::net’
         muduo::net::Buffer *buf;
                
         muduo::net::Buffer *buf;

**Comment by chenshuo** [comment]
Did you

```C
#include "muduo/net/Buffer.h"
```

?
